beep/featurize.py

```diff
@@ -47,6 +47,357 @@
 s = {'service': 'DataAnalyzer'}
 
 
+class BeepFeatures(MSONable):
```

Comment:
I would use the abc module here, e. g. make BeepFeatures an ABC or ABCMeta, use @abstractclassmethod and @abstractmethod for method handles.  Other than that looks good.